### ISSUE TITLE
ROL: Use of deprecated Tpetra code

### ISSUE THREAD

**Issue by tjfulle** [trilinos/Trilinos#5540]
## Bug Report
@trilinos/rol 

Relates to: #5536

### Description
Several of ROL's examples use the deprecated `Tpetra::CrsMatrix<>::getNode` and `clone` methods:

```
$ cd packages/rol/example
$ grep -irn 'getNode(' .
./PDE-OPT/0ld/TOOLS/PDE_FEM.hpp:669:    //ROL::Ptr<Tpetra::Details::DefaultTypes::node_type> node = matA_->getNode();
./PDE-OPT/0ld/TOOLS/PDE_FEM.hpp:723://    //ROL::Ptr<Tpetra::Details::DefaultTypes::node_type> node = matA_->getNode();
./PDE-OPT/0ld/adv-diff-react/data.hpp:532:    ROL::Ptr<Tpetra::Details::DefaultTypes::node_type> node = matA_->getNode();
./PDE-OPT/0ld/poisson/data.hpp:453:    ROL::Ptr<Tpetra::Details::DefaultTypes::node_type> node = matA_->getNode();
./PDE-OPT/0ld/stefan-boltzmann/data.hpp:475:    ROL::Ptr<Tpetra::Details::DefaultTypes::node_type> node = matA_->getNode();
./PDE-OPT/0ld/stefan-boltzmann/data.hpp:734:    ROL::Ptr<Tpetra::Details::DefaultTypes::node_type> node = matA_->getNode();
./PDE-OPT/0ld/stoch-adv-diff/data.hpp:605:    ROL::Ptr<Tpetra::Details::DefaultTypes::node_type> node = matA_->getNode();
./PDE-OPT/0ld/stoch-adv-diff/data.hpp:808:    ROL::Ptr<Tpetra::Details::DefaultTypes::node_type> node = matA_->getNode();
```

**Comment by tjfulle** [comment]
`getNode` is only used to cal (the deprecated) `Tpetra::CrsMatrix<>::clone`.  `clone` appears to be used to make a deep copy of the matrix.  The following diff gets around the compile error:

```
-    ROL::Ptr<Tpetra::Details::DefaultTypes::node_type> node = matA_->getNode();
-    matA_dirichlet_ = matA_->clone(node);
+    matA_dirichlet_ = ROL::makePtr<Tpetra::CrsMatrix<>>(*matA_, Teuchos::DataAccess::Copy);
```

apply this (or similar) everywhere clone is called.

**Comment by kyungjoo-kim** [comment]
@tjfulle I just compiled the code with deprecated off and inst int int ON. It compiled ROL examples and tests. Could you give me a more specific Tpetra configuration  ? I tested the default one too but it does not cause a compile problem.

**Comment by kyungjoo-kim** [comment]
Do I need to turn on warning as an error  ?

**Comment by tjfulle** [comment]
@kyngjoo-kim, see https://github.com/trilinos/Trilinos/issues/5602#issuecomment-516594843 for the config.  It could be that this has already been fixed by the ROL developers?

**Comment by tjfulle** [comment]
@kyungjoo-kim note, the config linked in the last comment disables ROL's examples, you would need to remove that line.

**Comment by kyungjoo-kim** [comment]
@tjfulle now I see some errors..

**Comment by kyungjoo-kim** [comment]
Hi @tjfulle I pushed into the issue-5536 branch. At least it compiles with deprecation code off. I also found that you already finish all changes related to getNode and clone. My fix is on the mismatch between local use of int type and Tpetra global ordinal type.

**Comment by tjfulle** [comment]
closed by #5739
